Klika-Tech/aws-greengrass-client-device-auth
# feat(uat): implement GGMQ scenario based on GGAD-1-T26

**Issue #, if available:**
https://klika-tech.atlassian.net/browse/GGMQ-273
Implement GGMQ scenario based on GGAD-1-T26

**Description of changes:**
- Add GGMQ-1-T26 scenario based on GGAD-1-T26
- Add disconnection check to GGMQ-1-T25
- Add disconnect type of event to event storage
- Add two steps to verify a certificate of broker
- Add two steps to check is connection disconnected or is not
- Replace RuntimeException with IllegalStateException in implemenetation of steps
- Rename steps to clear event storage
- Update readme by add sentense about firewall

**Why is this change necessary:**
We should port GGAD-1-T26 scenario to OTF base

**How was this change tested:**
By running Cucumber's scenarios

**Test results:**
```
[INFO ] 2023-08-02 19:06:50.952 [main] StepTrackingReporting - Passed: 'GGMQ-1-T26-v3-sdk-java: As a customer, my GGAD stays connected when CDA rotates cert in EMQX broker'
[INFO ] 2023-08-02 19:06:50.952 [main] StepTrackingReporting - Passed: 'GGMQ-1-T26-v3-mosquitto-c: As a customer, my GGAD stays connected when CDA rotates cert in EMQX broker'
[INFO ] 2023-08-02 19:06:50.953 [main] StepTrackingReporting - Passed: 'GGMQ-1-T26-v3-paho-java: As a customer, my GGAD stays connected when CDA rotates cert in EMQX broker'
[INFO ] 2023-08-02 19:06:50.953 [main] StepTrackingReporting - Passed: 'GGMQ-1-T26-v3-paho-python: As a customer, my GGAD stays connected when CDA rotates cert in EMQX broker'
[INFO ] 2023-08-02 19:06:50.953 [main] StepTrackingReporting - Passed: 'GGMQ-1-T26-v5-sdk-java: As a customer, my GGAD stays connected when CDA rotates cert in EMQX broker'
[ERROR] 2023-08-02 19:06:50.953 [main] StepTrackingReporting - Failed: 'GGMQ-1-T26-v5-mosquitto-c: As a customer, my GGAD stays connected when CDA rotates cert in EMQX broker': Failed at 'the Greengrass deployme
nt is COMPLETED on the device after 5 minutes'
[INFO ] 2023-08-02 19:06:50.953 [main] StepTrackingReporting - Passed: 'GGMQ-1-T26-v5-paho-java: As a customer, my GGAD stays connected when CDA rotates cert in EMQX broker'
[INFO ] 2023-08-02 19:06:50.953 [main] StepTrackingReporting - Passed: 'GGMQ-1-T26-v5-paho-python: As a customer, my GGAD stays connected when CDA rotates cert in EMQX broker'
```
and when one failed restarted:
```
[INFO ] 2023-08-02 20:05:39.241 [main] StepTrackingReporting - Passed: 'GGMQ-1-T26-v5-mosquitto-c: As a customer, my GGAD stays connected when CDA rotates cert in EMQX broker'
```

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
